python/ray/tune/trial.py

```diff
@@ -48,6 +50,61 @@ def __str__(self):
             return "{}:{}".format(self.hostname, self.pid)
 
 
+class TrialDirSchema(object):
+    """Describes the Trial's directory structure.
+
+    {trial_dir_name}/
+      driver.log.1
+      driver.log.2
+      remote_logs/
+      checkpoints/
+    """
+    REMOTE_LOGDIR = "remote_logs"
+    CHECKPOINT_DIR = "checkpoints"
+
+    def __init__(self, trial_name, local_dir):
+        local_dir = os.path.expanduser(local_dir)
+        if not os.path.exists(local_dir):
+            os.makedirs(local_dir)
+        self.root_dir = tempfile.mkdtemp(
+            prefix="{}_{}".format(trial_name[:MAX_LEN_IDENTIFIER], date_str()),
+            dir=local_dir)
+        self.mkdir()
+
+    def mkdir(self):
+        if not os.path.exists(self.root_dir):
+            os.makedirs(self.root_dir)
+        if not os.path.exists(self.logdir):
+            os.makedirs(self.logdir)
+        if not os.path.exists(self.remote_logdir):
+            os.makedirs(self.remote_logdir)
+        if not os.path.exists(self.checkpoint_dir):
+            os.makedirs(self.checkpoint_dir)
+
+    @property
+    def root(self):
+        """Root directory."""
+        return self.root_dir
+
+    @property
+    def logdir(self):
+        """Directory containing logs that originated on the driver.
+
+        For backwards compatibility, this is set to the root.
```

Comment:
So the backwards compat is needed because users have weird scripts written to navigate and parse the outputs assuming this particular directory structure. 

Now if that wasn't an issue, I'm not sure if there's any difference in putting the driver logs in a specific folder, but maybe I'm wrong? @ujvl.

Reply:
yeah there's no real benefit from placing it in a subdir, aside from clarity